Remove deprecated uppercase URL attribute alias on WebSocket interface

Ahmad-S792 opened the pull request:
#### a05af340527ab26ff2e81de10b6630b25b83e922
<pre>
Remove deprecated uppercase URL attribute alias on WebSocket interface

Remove deprecated uppercase URL attribute alias on WebSocket interface
<a href="https://bugs.webkit.org/show_bug.cgi?id=248781">https://bugs.webkit.org/show_bug.cgi?id=248781</a>

Reviewed by Chris Dumez.

This patch is to align Webkit with Blink / Chromium, Gecko / Firefox and Web-Specification:

<a href="https://html.spec.whatwg.org/multipage/comms.html#the-websocket-interface">https://html.spec.whatwg.org/multipage/comms.html#the-websocket-interface</a>

Partial Merge - <a href="https://chromium.googlesource.com/chromium/blink/+/825c9732ce83f380fadfc062331cc12495e5eb12">https://chromium.googlesource.com/chromium/blink/+/825c9732ce83f380fadfc062331cc12495e5eb12</a>

The `URL` attribute is from early iterations of the spec and was
maintained for compatibility even after `url` was standardized but now it is not in the latest web-specification and was removed from Blink in 2015 as well.

* Source/WebCore/Modules/websockets/WebSocket.idl: Remove &apos;URL&apos; attribute and also add link to web-spec
* LayoutTests/http/tests/websocket/tests/hybi/network-process-crash-error.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/simple.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-attribute.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-attribute-expected.txt: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-parsing.html: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/url/004-expected.txt: Rebaselined
* LayoutTests/http/tests/websockets/tests/hybi/url-parsing-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/url/004-expected.txt: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/simple-wss.html: Rebaselined

Canonical link: <a href="https://commits.webkit.org/257467@main">https://commits.webkit.org/257467@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/edb1b6735651bb1c69f619a7184cb580db280e96

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99007 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8213 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32160 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108413 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168665 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102969 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8766 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85551 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91530 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106371 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104727 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6636 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90219 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33661 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88458 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21564 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76515 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2114 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23080 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2018 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45470 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5143 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6981 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42555 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3432 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
Blink and Gecko don't support `URL` so LGTM.

Ahmad-S792 committed:
Remove deprecated uppercase URL attribute alias on WebSocket interface

Remove deprecated uppercase URL attribute alias on WebSocket interface
https://bugs.webkit.org/show_bug.cgi?id=248781

Reviewed by Chris Dumez.

This patch is to align Webkit with Blink / Chromium, Gecko / Firefox and Web-Specification:

https://html.spec.whatwg.org/multipage/comms.html#the-websocket-interface

Partial Merge - https://chromium.googlesource.com/chromium/blink/+/825c9732ce83f380fadfc062331cc12495e5eb12

The `URL` attribute is from early iterations of the spec and was
maintained for compatibility even after `url` was standardized but now it is not in the latest web-specification and was removed from Blink in 2015 as well.

* Source/WebCore/Modules/websockets/WebSocket.idl: Remove 'URL' attribute and also add link to web-spec
* LayoutTests/http/tests/websocket/tests/hybi/network-process-crash-error.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/simple.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-attribute.html: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-attribute-expected.txt: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/url-parsing.html: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/url/004-expected.txt: Rebaselined
* LayoutTests/http/tests/websockets/tests/hybi/url-parsing-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/url/004-expected.txt: Rebaselined
* LayoutTests/http/tests/websocket/tests/hybi/simple-wss.html: Rebaselined

Canonical link: https://commits.webkit.org/257467@main
